### DISCUSSION TITLE
Is there any similar method like secretsmanager:BatchGetSecretValue in AWS SDK v1.x?

### DISCUSSION THREAD

**Discussion by asad-miah** [aws/aws-sdk-java#3088] [Q&A] [answered]
I am using AWS SDK 1.12.97. I need to use secretsmanager:BatchGetSecretValue method in my project. But it seems that the method is not available in version 1.x. Is there any similar method in AWS SDK v1.x like batchGetSecretValue()? If not, is it possible to use the AWS SDK V2.x and AWS SDK V1.x in the same maven project at the same time?

**Comment by debora-ito** [comment, marked as answer]
BatchGetSecretValue was added in SDK version 1.12.596, you'd need to upgrade to this version at least.

The easiest way to check in which version something was added to the SDK is to use Git "blame" - for example, here's the Github UI with blame information on the left side :-
https://github.com/aws/aws-sdk-java/blame/master/aws-java-sdk-secretsmanager/src/main/java/com/amazonaws/services/secretsmanager/AWSSecretsManagerClient.java

> If not, is it possible to use the AWS SDK V2.x and AWS SDK V1.x in the same maven project at the same time?

Yes it is possible. We have more information on how to achieve this in the Java SDK v2 Developer Guide - https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-side-by-side.html

**Reply by asad-miah** [reply]
Thanks a lot for the answer @debora-ito
